### DISCUSSION TITLE
Help generating a bootable Ubuntu bionic

### DISCUSSION THREAD

**Discussion by YanickNoblanc** [systemd/mkosi#2187] [Q&A] [answered]
Hi all

I am trying to generate an Ubuntu bionic bootable on bare-metal and testable with qemu.
I am working on a Debian Trixie VirtualBox VM where mkosi 19 (package) is installed.
I cannot figure out mkosi options to set, to generate this bootable image. Here is the configuration I have been trying to use so far:

Files:
```
.
|-- mkosi.conf
|-- mkosi.repart
|   |-- 00-esp.conf
|   `-- 10-root.conf
`-- mkosi.skeleton
    `-- etc
        |-- apt
        |   |-- apt.conf.d
        |   |   `-- 95-mesta-repo
        |   |-- sources.list
        |   `-- sources.list.d
        |       |-- mesta-compact.list
        |       `-- ubuntu-bionic-amd64.sources
        |-- hostname
        `-- hosts

```

File mkosi.conf:
```
[Distribution]
Distribution=ubuntu
Release=bionic
Mirror=https://<url to our ubuntu internal mirrors>

[Output]
Format=disk

[Content]
Packages=linux-image-generic,systemd,udev,dbus,kmod,apt,netplan.io,sudo,nano,wget,gpg,pciutils,fdisk,ca-certificates,apt-utils,iputils-ping,init-system-helpers=1.56+nmu1~ubuntu18.04.1,usbutils

Autologin=yes
RootPassword=****
Bootable=yes
#Bootloader=grub    #If set, mkosi systematical fails saying grub for EFI was not installed in /efi
```

File mkosi.repart/00-esp.conf:
```
[Partition]
Type=esp
Format=vfat
CopyFiles=/efi:/
SizeMinBytes=512M
SizeMaxBytes=512M
```

File mkosi.repart/10-root.conf:
```
[Partition]
Type=root
Format=ext4
CopyFiles=/
SizeMinBytes=4G
SizeMaxBytes=4G
```

Files in mkosi.skeleton/etc/apt/ : We are using internal ubuntu mirrors rather than the official ones, therefore declaring those in our sources.list. The host is also pointing to those mirrors when executing mkosi.

mkosi command executed :  mkosi --force --directory='path to mkosi files' --output utl-infra qemu

The image building by mkosi is successful, but when qemu tries launching the image it systematical fails finding a suitable root device.
```
	[    2.431988] VFS: Cannot open root device "(null)" or unknown-block(0,0): error -6
	[    2.444733] Please append a correct "root=" boot option; here are the available partitions:
	[    2.454787] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)

```

Could someone please tell me what is wrong or missing in my config/commands (Packages installed, Partitions, Bootloader, Format, ... ?)
Or detail the proper way(s) to generate an Ubuntu bionic image bootable on bare-metal ?

Thank you very much for your help

**Comment by daandemeyer** [comment, marked as answer]
Our default compression for Ubuntu initrds is zstd and the bionic 18 kernel cannot decompress those. Given that bionic is EOL, we're not going to add a workaround in mkosi for this. You can fork mkosi and change `config_default_compression()` to return `Compression.xz` if the distribution is ubuntu and the release is bionic.

**Comment by YanickNoblanc** [comment]
Thank you very much for the quick feedback. Will I get same compression issues with more recent versions of Ubuntu (focal, jammy, ...) ?

**Reply by daandemeyer** [reply]
Jammy won't have the issue, not sure about focal

**Comment by YanickNoblanc** [comment]
Thank you so much for your help
